babi/screen.py

```diff
@@ -518,8 +517,15 @@ def save(self) -> PromptResult | None:
             ) as f:
                 f.write(contents)
         except OSError as e:
-            self.status.update(f'cannot save file: {e}')
-            return PromptResult.CANCELLED
+            if e.errno == 2:
+                os.makedirs(os.path.dirname(self.file.filename))
```

Comment:
ah, I told you the correct code in the previous review -- this still isn't correct -- see also:
- what is atomicity? https://youtu.be/tmlmCG5egI0
- what is "idempotent"? https://youtu.be/o9THkT5ZPi4